src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,118 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold = 12;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = Math.abs(hash(key));
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        if (table[index] != null) {
+            for (Node<K, V> currentNode = table[index]; currentNode != null;
+                    currentNode = currentNode.next) {
+
+                if ((currentNode.hash == newNode.hash && currentNode.key == newNode.key)
```

Comment:
The condition `currentNode.key == newNode.key` is not necessary because `Objects.equals` already checks for reference equality before checking for object equality.